sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -1578,7 +1574,7 @@ def get_blob_client(
                 DeprecationWarning
             )
         try:
-            blob_name = blob.get('name')
+            blob_name = blob.get('name')  # type: ignore
```

Comment:
We will likely have to keep this to maintain the typehint as `str` to dissaude people from using the "deprecated" `BlobProperties`